tests/test_uncertain_data.py

```diff
@@ -93,6 +93,28 @@ def test_scalardist(self):
         data2 = pickle.load(open('test.pkl', 'rb'))
         self.assertEqual(d, data2)
 
+    def test_pickle_unweightedsamples(self):
```

Comment:
I just noticed the pick tests already in the file (e.g., line 93-94). I think it makes sense to have it as separate tests like you do. Could you remove the earlier ones from the file?